Content.Server/AI/Operators/Inventory/InteractWithEntityOperator.cs

```diff
@@ -0,0 +1,48 @@
+using Content.Server.GameObjects.Components.Mobs;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Utility;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+
+namespace Content.Server.AI.Operators.Inventory
+{
+    /// <summary>
+    /// A Generic interacter; if you need to check stuff then make your own
+    /// </summary>
+    public class InteractWithEntityOperator : AiOperator
+    {
+        private readonly IEntity _owner;
+        private readonly IEntity _useTarget;
+
+        public InteractWithEntityOperator(IEntity owner, IEntity useTarget)
+        {
+            _owner = owner;
+            _useTarget = useTarget;
+
+        }
+
+        public override Outcome Execute(float frameTime)
+        {
+            if (_useTarget.Transform.GridID != _owner.Transform.GridID)
+            {
+                return Outcome.Failed;
+            }
+
+            if (InteractionChecks.InRangeUnobstructed(_owner, _useTarget.Transform.MapPosition))
+            {
+                return Outcome.Failed;
+            }
```

Comment:
Did you not mean to negate these checks?

Reply:
I'm an idiot, I fixed it. I also spotted another one I missed in the operators.